crates/sui-replay-2/src/tracing.rs

```diff
@@ -4,89 +4,39 @@
-const TRACE_FILE_NAME: &str = "trace.json.zst";
 const BCODE_DIR: &str = "bytecode";
 const SOURCE_DIR: &str = "source";
 
-/// Gets the path to store trace output (either the default one './replay' or user-specified).
-/// Upon success, the path will exist in the file system.
-pub fn get_trace_output_path(trace_execution: Option<PathBuf>) -> Result<PathBuf, anyhow::Error> {
-    match trace_execution {
-        Some(path) => {
-            if !path.exists() {
-                return Err(anyhow::anyhow!(format!(
-                    "User-specified path to store trace output does not exist: {:?}",
-                    path
-                )));
-            }
-            if !path.is_dir() {
-                return Err(anyhow::anyhow!(format!(
-                    "User-specified path to store trace output is not a directory: {:?}",
-                    path
-                )));
-            }
-            Ok(path)
-        }
-        None => {
-            let current_dir = env::current_dir().context("Failed to get current directory")?;
-            let path = current_dir.join(DEFAULT_TRACE_OUTPUT_DIR);
-            if path.exists() && path.is_file() {
-                return Err(anyhow::anyhow!(
-                    format!(
-                        "Default path to store trace output already exists and is a file, not a directory: {:?}",
-                        path
-                    )
-                ));
-            }
-            fs::create_dir_all(&path).context("Failed to create default trace output directory")?;
-            Ok(path)
-        }
-    }
-}
-
 /// Saves the trace and additional metadata needed to analyze the trace
 /// to a subderectory named after the transaction digest.
 pub fn save_trace_output(
-    output_path: &Path,
-    digest: &str,
+    artifact_manager: &ArtifactManager<'_>,
     trace_builder: MoveTraceBuilder,
     context_and_effects: &TxnContextAndEffects,
 ) -> Result<(), anyhow::Error> {
-    let txn_output_path = output_path.join(digest);
-    if txn_output_path.exists() {
-        return Err(anyhow::anyhow!(
-            "Trace output directory for transaction {} already exists: {:?}",
-            digest,
-            txn_output_path,
-        ));
-    }
-    fs::create_dir_all(&txn_output_path).context(format!(
-        "Failed to create trace output directory for transaction {}",
-        digest,
-    ))?;
     let trace = trace_builder.into_trace();
-    let json = trace.into_compressed_json_bytes();
-    let trace_file_path = txn_output_path.join(TRACE_FILE_NAME);
-    fs::write(&trace_file_path, json).context(format!(
-        "Failed to write trace output to {:?}",
-        trace_file_path,
-    ))?;
-    let bcode_dir = txn_output_path.join(BCODE_DIR);
+    let trace_member = artifact_manager.member(Artifact::Trace);
+    trace_member
+        .serialize_move_trace(trace)
+        .transpose()?
+        .unwrap();
+
+    // TODO: have this use the artifact manager as well.
```

Comment:
Not sure how much sense it would make to make it into separate artifacts as these are mostly usable along the trace file. Of course I am writing this not really knowing what the plan here would be :-) At the very least, I think these should keep getting generated with the trace by default (so that you don't need to command-line options to get both the trace and this additional data needed for trace debugging)

Reply:
Agreed that from a user/CLI facing viewpoint that you shouldn't need to specify one or the other (and strongly agreed that we should keep it that way). I think the main thing is what we use underneath the hood/here to output and manage the different artifacts here. Right now I think I might keep this as-is in this PR, but will followup with a separate PR to allow a bit more of a rich structure to the output artifacts (which packages require).